File: TaskTracker.Core/src/Models/PostRequests/CreateOrEditWorkSpacePostRequest.cs
using System.Globalization;
using TaskTracker.Core.src.Constants;
using TaskTracker.Core.src.Enums;

namespace TaskTracker.Core.src.Models.PostRequests
{
    public class CreateOrEditWorkspacePostRequest : BasePostRequest
    {
        public int Id { get; set; }
        /// <summary>
        /// Название рабочего пространства
        /// </summary>
        public string Name { get; set; } = null!;
        public WorkspaceType WorkspaceType { get; set; }

        /// <summary>
        /// Ссылка на управляющего компании
        /// </summary>
        public int? DirectorUserId { get; set; }

        //Все поля ниже заполняются, если WorkspaceType - Company

        /// <summary>
        /// Страна, в которой компания зарегистрирована
        /// </summary>
        public int? Country { get; set; }

        /// <summary>
        /// Дата регистрации в UTC
        /// </summary>
        public string? RegistrationDate { get; set; }

        /// <summary>
        /// Юр. адрес
        /// </summary>
        public string? Address { get; set; }

        public string? INN { get; set; }

        public DateTime? RegistrationDateTime
        { 
            get
            {
                return !string.IsNullOrEmpty(RegistrationDate) ? DateTime.ParseExact(RegistrationDate, DateFormatConstants.FrontInputFormat, CultureInfo.InvariantCulture) : null;
            } 
        }

        public WorkspaceReviewStatus? ReviewStatus { get; set; }
    }
}

File: ui/task-tracker/src/app/shared/model/postRequests/createOrEditWorkSpacePostRequest.ts
import { WorkspaceType } from "../../enums/work-space-type";
import { WorkspaceReviewStatus } from "../../enums/workspace-review-status";

export class CreateOrEditWorkspacePostRequest{

    public id : number | undefined;
    
    /**  Название рабочего пространства */
    public name! : string;
    
    public workspaceType! : WorkspaceType;
    
    /** Ссылка на управляющего компании */
    public directorUserId : number | undefined;
    
    //Все поля ниже заполняются, если WorkspaceType - Company
    
    /** Страна, в которой компания зарегистрирована */
    public country : number | undefined;
    
    /** Дата регистрации в UTC */
    public registrationDate : string | undefined;
    
    /** Юр. адрес */
    public address : string | undefined;
    
    public iNN : number | undefined;

    public reviewStatus : WorkspaceReviewStatus | undefined;
}
